docs/julia.rst

```diff
@@ -0,0 +1,399 @@
+        all functions and variables declared in the block (but not its sub-blocks)
+        variables are marked inactive for now
+        TODO: more formal
+        G1, L'1 = E(G, L', St1)
+        G2, L'2 = E(G1, L'1, St2)
+        ...
+        Gn, L'n = E(G(n-1), L'(n-1), Stn)
+        let L'' be a copy of L'n where the innermost scope is removed
+        Gn, L''
+    E(G, L, <function fname (param1, ..., paramn) -> (ret1, ..., retm) block>: FunctionDefinition) =
+        G, L
+    E(G, L, <let (var1, ..., varn) := value>: VariableDeclaration) =
+        E(G, L, <(var1, ..., varn) := value>: Assignment)
+    E(G, L, <(var1, ..., varn) := value>: Assignment) =
+        let G', L', v1, ..., vn = E(G, L, value)
+        let L'' be a copy of L' where L'[vi] = vi for i = 1, ..., n
+        G, L''
+    E(G, L, name: Identifier) =
+        G, L, L[name]
+    E(G, L, fname(arg1, ..., argn): FunctionCall) =
+        G1, L1, vn = E(G, L, argn)
+        ...
+        G(n-1), L(n-1), v2 = E(G(n-2), L(n-2), arg2)
+        Gn, Ln, v1 = E(G(n-1), L(n-1), arg1)
+        Let <function fname (param1, ..., paramn) -> (ret1, ..., retm) block>
+        be the function L[fname].
+        Let L' be a copy of L that does not contain any variables in any scope,
+        but which has a new innermost scope such that
+        L'[parami] = vi and L'[reti] = 0
+        Let G'', L'', rv1, ..., rvm = E(Gn, L', block)
+        G'', Ln, rv1, ..., rvm
+    E(G, L, l: HexLiteral) = G, L, hexString(l),
+        where hexString decodes l from hex and left-aligns in into 32 bytes
+    E(G, L, l: StringLiteral) = G, L, utf8EncodeLeftAligned(l),
+        where utf8EncodeLeftAligned performs a utf8 encoding of l
+        and aligns it left into 32 bytes
+    E(G, L, n: HexNumber) = G, L, hex(n)
+        where hex is the hexadecimal decoding function
+    E(G, L, n: DecimalNumber) = G, L, dec(n),
+        where dec is the decimal decoding function
+
+Type Conversion Functions
+-------------------------
+
+JULIA has no support for implicit type conversion and therefore functions exists to provide explicit conversion.
+When converting a larger type to a shorter type a runtime exception can occur in case of an overflow.
+
+The following type conversion functions must be available:
+- ``u32tobool(x:u32) -> (y:bool)``
+- ``booltou32(x:bool) -> (y:u32)``
+- ``u32tou64(x:u32) -> (y:u64)``
+- ``u64tou32(x:u64) -> (y:u32)``
+- etc. (TBD)
+
+Low-level Functions
+-------------------
+
+The following functions must be available:
+
++---------------------------------------------------------------------------------------------------------------+
+| *Arithmetics*                                                                                                 |
++---------------------------------------------------------------------------------------------------------------+
+| addu256(x:u256, y:u256) -> z:u256           | x + y                                                           |
```

Comment:
That's kind-of inconsistent with our choice of `a, b := f()` without parentheses on the left-hand side.